doc/source/index.rst

```diff
@@ -85,8 +85,194 @@ should have.
+         vnet_name=vnet_name,
+         subnet_name=subnet_name,
+         idle_seconds_before_scaledown=300,
+
+         ### UNCOMMENT BELOW FOR LOCAL RUNS
+         # admin_username=admin_username,
+         # admin_user_ssh_key=ssh_key_pub,
+         # remote_login_port_public_access='Enabled',
+      )
+      ct = ComputeTarget.create(ws, ct_name, config)
+      ct.wait_for_completion(show_output=True)
+   else:
+      ct = ws.compute_targets[ct_name]
+
+If your compute target already exists you can call ``ct = ws.compute_targets[ct_name]``.
+
+Setting up vnet
+"""""""""""""""
+If you do not have a VNET created there are two ways to create one.
+
+1. Using `https://azure.portal.com <https://azure.portal.com>`_:
+
+   a. On the home page click on `+ Create a resource` on the top-left portion of the page.
+   b. Search for `Virtual Network`.
+   c. Click on `Create` and follow the instructions: select the `Subscription` and the 
+      `Resource group` you will create the vnet in, and provide a `Name` and the `Location`. 
+      **NOTE:** keep the location the same as your AzureML Workspace e.g. if your AzureML Workspace 
+      location is `eastus` you should create the Virtual Network in `East US`.
+   d. Click `Review+Create` and follow through with the rest of the instructions.
+
+2. Using Azure CLI:
+
+   a. Install the AzureCLI; instructions are here: 
+      `https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest 
+      <https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest>`_
+   b. Open terminal and login to your Azure Subscription: ``az login``. This should 
+      automatically log you in into your Azure subscription. **NOTE:** If you have more than 
+      one subscription you will need to set the right subscription to use:
+      ``az account set --subscription "<name-of-sub-to-use>"``
+   c. Create the virtual network: 
+      ``az network vnet create -g <resource-group> -n <vnet-name> --location <location> --subnet-name default``
+
+Define Environment
+~~~~~~~~~~~~~~~~~~
+
+For the `Environment <https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.environment.environment?view=azure-ml-py>`_  
+we will use a AzureML curated environment for running Dask CPU cluster, the `AzureML-Dask-CPU`. 
+However, the ``Environment`` class allows you to specify your own docker image and additional packages to install.
+
+.. code-block:: python
+
+   packages = ["matplotlib"]
+
+   env = Environment(name=env_name)
+
+   for package in packages:
+      env.python.conda_dependencies.add_pip_package(package)
+
+Create cluster
+~~~~~~~~~~~~~~
+
+To create cluster:
+
+.. code-block:: python
+
+   amlcluster = AzureMLCluster(
+      workspace=ws,
+      compute_target=ct,
+      environment_definition=env,
+      initial_node_count=2,
+
+      ### UNCOMMENT BELOW FOR LOCAL RUNS
+      # admin_username=admin_username,
+      # admin_ssh_key=admin_ssh_key_priv,
+   )
```

Comment:
I've managed to run through to this point but now I'm getting this exception.

```
############################## Setting up cluster ##############################
########################## Submitting the experiment ###########################

WARNING - This compute target type doesn't support non-Docker runs; overriding run configuration enable Docker.

---------------------------------------------------------------------------
TrainingException                         Traceback (most recent call last)
<ipython-input-10-b4391ff8aa41> in <module>
      6     ### UNCOMMENT BELOW FOR LOCAL RUNS
      7     admin_username=admin_username,
----> 8     admin_ssh_key=admin_ssh_key_priv,
      9 )

~/Projects/dask/dask-cloudprovider/dask_cloudprovider/providers/azure/azureml.py in __init__(self, workspace, compute_target, environment_definition, experiment_name, initial_node_count, jupyter, jupyter_port, dashboard_port, scheduler_port, additional_ports, admin_username, admin_ssh_key, datastores, code_store, asynchronous, **kwargs)
    222             self._loop_runner.start()
    223             self.sync(self.__get_defaults)
--> 224             self.sync(self.__create_cluster)
    225 
    226     async def __get_defaults(self):

~/Projects/dask/distributed/distributed/deploy/cluster.py in sync(self, func, asynchronous, callback_timeout, *args, **kwargs)
    159             return future
    160         else:
--> 161             return sync(self.loop, func, *args, **kwargs)
    162 
    163     async def _get_logs(self, scheduler=True, workers=True):

~/Projects/dask/distributed/distributed/utils.py in sync(loop, func, callback_timeout, *args, **kwargs)
    346     if error[0]:
    347         typ, exc, tb = error[0]
--> 348         raise exc.with_traceback(tb)
    349     else:
    350         return result[0]

~/Projects/dask/distributed/distributed/utils.py in f()
    330             if callback_timeout is not None:
    331                 future = asyncio.wait_for(future, callback_timeout)
--> 332             result[0] = yield future
    333         except Exception as exc:
    334             error[0] = sys.exc_info()

~/miniconda3/envs/dask/lib/python3.7/site-packages/tornado/gen.py in run(self)
    733 
    734                     try:
--> 735                         value = future.result()
    736                     except Exception:
    737                         exc_info = sys.exc_info()

~/Projects/dask/dask-cloudprovider/dask_cloudprovider/providers/azure/azureml.py in __create_cluster(self)
    341         )
    342 
--> 343         run = exp.submit(estimator)
    344 
    345         self.__print_message("Waiting for scheduler node's IP")

~/miniconda3/envs/dask/lib/python3.7/site-packages/azureml/core/experiment.py in submit(self, config, tags, **kwargs)
    200         submit_func = get_experiment_submit(config)
    201         with self._log_context("submit config {}".format(config.__class__.__name__)):
--> 202             run = submit_func(config, self.workspace, self.name, **kwargs)
    203         if tags is not None:
    204             run.set_tags(tags)

~/miniconda3/envs/dask/lib/python3.7/site-packages/azureml/train/_estimator_helper.py in _estimator_submit_method(estimator, workspace, experiment_name, **kwargs)
     67             source_directory_data_store=source_directory_data_store)
     68 
---> 69     experiment_run = estimator._fit(workspace, experiment_name)
     70 
     71     if override_params:

~/miniconda3/envs/dask/lib/python3.7/site-packages/azureml/train/estimator/_mml_base_estimator.py in _fit(self, workspace, experiment_name)
    152         self._last_submitted_runconfig = self.run_config
    153 
--> 154         return self._submit(workspace, experiment_name, telemetry_values)
    155 
    156     def _override_params(self, script_params=None, inputs=None, source_directory_data_store=None):

~/miniconda3/envs/dask/lib/python3.7/site-packages/azureml/train/estimator/_mml_base_estimator.py in _submit(self, workspace, experiment_name, telemetry_values)
    146                 return experiment_run
    147             except AzureMLException as e:
--> 148                 raise TrainingException(e.message, inner_exception=e) from None
    149 
    150     def _fit(self, workspace, experiment_name):

TrainingException: TrainingException:
	Message: {
    "error_details": {
        "correlation": {
            "operation": "a6a7323f7fafad4d95f1355d15f54b9b",
            "request": "7ae8ab794c8df3f7"
        },
        "environment": "eastus2",
        "error": {
            "code": "UserError",
            "message": "Environment name can not start with the prefix AzureML. To alter a curated environment first create a copy of it."
        },
        "location": "eastus2",
        "time": "2020-03-24T20:33:25.0813057+00:00"
    },
    "status_code": 400,
    "url": "https://eastus2.experiments.azureml.net/execution/v1.0/subscriptions/fc4f4a6b-4041-4b1c-8249-854d68edcf62/resourceGroups/nvaiinfra-rg/providers/Microsoft.MachineLearningServices/workspaces/rapids-dask-test/experiments/dask-cloudprovider/startrun?runId=dask-cloudprovider_1585081998_7db240ee"
}
	InnerException ExperimentExecutionException:
	Message: {
    "error_details": {
        "correlation": {
            "operation": "a6a7323f7fafad4d95f1355d15f54b9b",
            "request": "7ae8ab794c8df3f7"
        },
        "environment": "eastus2",
        "error": {
            "code": "UserError",
            "message": "Environment name can not start with the prefix AzureML. To alter a curated environment first create a copy of it."
        },
        "location": "eastus2",
        "time": "2020-03-24T20:33:25.0813057+00:00"
    },
    "status_code": 400,
    "url": "https://eastus2.experiments.azureml.net/execution/v1.0/subscriptions/fc4f4a6b-4041-4b1c-8249-854d68edcf62/resourceGroups/nvaiinfra-rg/providers/Microsoft.MachineLearningServices/workspaces/rapids-dask-test/experiments/dask-cloudprovider/startrun?runId=dask-cloudprovider_1585081998_7db240ee"
}
	InnerException None
	ErrorResponse 
{
    "error": {
        "message": "{\n    \"error_details\": {\n        \"correlation\": {\n            \"operation\": \"a6a7323f7fafad4d95f1355d15f54b9b\",\n            \"request\": \"7ae8ab794c8df3f7\"\n        },\n        \"environment\": \"eastus2\",\n        \"error\": {\n            \"code\": \"UserError\",\n            \"message\": \"Environment name can not start with the prefix AzureML. To alter a curated environment first create a copy of it.\"\n        },\n        \"location\": \"eastus2\",\n        \"time\": \"2020-03-24T20:33:25.0813057+00:00\"\n    },\n    \"status_code\": 400,\n    \"url\": \"https://eastus2.experiments.azureml.net/execution/v1.0/subscriptions/fc4f4a6b-4041-4b1c-8249-854d68edcf62/resourceGroups/nvaiinfra-rg/providers/Microsoft.MachineLearningServices/workspaces/rapids-dask-test/experiments/dask-cloudprovider/startrun?runId=dask-cloudprovider_1585081998_7db240ee\"\n}"
    }
}
	ErrorResponse 
{
    "error": {
        "message": "{\n    \"error_details\": {\n        \"correlation\": {\n            \"operation\": \"a6a7323f7fafad4d95f1355d15f54b9b\",\n            \"request\": \"7ae8ab794c8df3f7\"\n        },\n        \"environment\": \"eastus2\",\n        \"error\": {\n            \"code\": \"UserError\",\n            \"message\": \"Environment name can not start with the prefix AzureML. To alter a curated environment first create a copy of it.\"\n        },\n        \"location\": \"eastus2\",\n        \"time\": \"2020-03-24T20:33:25.0813057+00:00\"\n    },\n    \"status_code\": 400,\n    \"url\": \"https://eastus2.experiments.azureml.net/execution/v1.0/subscriptions/fc4f4a6b-4041-4b1c-8249-854d68edcf62/resourceGroups/nvaiinfra-rg/providers/Microsoft.MachineLearningServices/workspaces/rapids-dask-test/experiments/dask-cloudprovider/startrun?runId=dask-cloudprovider_1585081998_7db240ee\"\n}"
    }
}
```

Reply:
My bad -- I put a wrong ENV definition. Submitting another push.